File: Easy/226. Invert Binary Tree/InvertBinaryTree.py
'''
LeetCode Easy: 226. Invert Binary Tree
https://leetcode.com/problems/invert-binary-tree/

Check InvertBinaryTree.cpp for brief solution explanation.
'''
class Solution:
    def dfs(self, root):
        if root==None:
            return
        self.dfs(root.left)
        self.dfs(root.right)
        root.left, root.right = root.right, root.left
    
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.dfs(root)
        return root

File: Easy/226. Invert Binary Tree/InvertBinaryTree.java
/*
LeetCode Easy: 226. Invert Binary Tree
https://leetcode.com/problems/invert-binary-tree/

Check InvertBinaryTree.cpp for brief solution explanation.
*/
class Solution {
    public void dfs(TreeNode root){
        if(root!=null){
            dfs(root.left);
            dfs(root.right);
            TreeNode tmp = root.left;
            root.left = root.right;
            root.right = tmp;
        }
    }
    public TreeNode invertTree(TreeNode root) {
        dfs(root);
        return root;
    }
}